src/compiler/checker.ts

```diff
@@ -9685,9 +9858,22 @@ namespace ts {
                         }
                     }
                 }
+                else if (source.flags & TypeFlags.Conditional) {
+                    const constraint = getConstraintOfDistributiveConditionalType(<ConditionalType>source);

```

Comment:
is it equivalent just to call `isRelatedTo` once on the result of getConstraintOfConditionalType, which ors together the two functions here?

Reply:
No, we want to check both for distributive conditional types.